### PR TITLE
Put special team/etc IDs together

### PR DESCRIPTION
**Pull request by harsha-mangena**
- refer #1532
- moved constants to `GlobalConstants.h` and segregated `MAX_AIS`

### CODE REVIEW

File: rts/ExternalAI/SSkirmishAICallbackImpl.cpp [hunk 1]
```diff
@@ -67,7 +67,7 @@ static inline CAICheats* GetCheatCallBack(int skirmishAIId) { return &AI_LEGACY_
 
 
 static void CheckSkirmishAIId(int skirmishAIId, const char* caller) {
-	if (skirmishAIId >= 0 && skirmishAIId < MAX_AIS)
+	if (skirmishAIId >= 0 && skirmishAIId < SPECIAL_AI_ID)
```
Comment (sprunk) [inline rts/ExternalAI/SSkirmishAICallbackImpl.cpp:70]: Uses related to normal IDs should stay using `MAX_AIS`

File: rts/Sim/Misc/GlobalConstants.h [hunk 2]
```diff
@@ -143,5 +144,26 @@ static constexpr float MAX_PROJECTILE_HEIGHT = 1e6f;
  */
 static constexpr int MAX_UNIT_SENSOR_RADIUS = 32768;
 
+
+ /**
+ * ID used to send chat messages to allies.
+ */
+static constexpr int TO_ALLIES = 252;
```
Comment (sprunk) [inline rts/Sim/Misc/GlobalConstants.h:151]: these would ideally be in a dense clump alongside `MAX_PLAYERS`

File: rts/ExternalAI/SSkirmishAICallbackImpl.cpp (hunk 1 above)
Comment (harsha-mangena) [reply rts/ExternalAI/SSkirmishAICallbackImpl.cpp]: changed to `MAX_AIS`

File: rts/Sim/Misc/GlobalConstants.h [hunk 3]
```diff
@@ -70,13 +70,34 @@ static constexpr int MAX_TEAMS = 255;
...
+/**
+ * ID send chat messages to everyone.
+ */
+static constexpr int TO_EVERYONE = 254;
+
+/**
+ * "player" number for GameServer-generated messages.
+ */
+static constexpr int SERVER_PLAYER = 255;
+
```
Comment (sprunk) [inline rts/Sim/Misc/GlobalConstants.h:92]: the names should be self-explanatory enough not to split them up with "this is a bridge" style docs
```suggestion
static constexpr int MAX_PLAYERS   = 251;
static constexpr int TO_ALLIES     = 252;
static constexpr int TO_SPECTATORS = 253;
static constexpr int TO_EVERYONE   = 254;
static constexpr int SERVER_PLAYER = 255;
```

File: rts/ExternalAI/SkirmishAIHandler.h [hunk 4]
```diff
@@ -201,7 +201,7 @@ class CSkirmishAIHandler
 	spring::unordered_set<std::string> luaAIShortNames;
 
 	// the current local AI ID that is executing, MAX_AIS if none (e.g. LuaUI)
```
Comment (sprunk) [inline rts/ExternalAI/SkirmishAIHandler.h:203]: comment looks incorrect now

Comment (harsha-mangena) [reply rts/ExternalAI/SkirmishAIHandler.h]: changed as per suggestion

**Review by sprunk — COMMENTED** [review]
https://github.com/beyond-all-reason/spring/pull/1536/files#r1635310537 still unresolved

File: rts/Net/GameServer.cpp [hunk 5]
```diff
@@ -1670,7 +1670,7 @@ void CGameServer::ProcessPacket(const unsigned playerNum, std::shared_ptr<const
 				}
 
 				// discard bogus ID from message, reserve actual slot here
-				if ((skirmishAIId = ReserveSkirmishAIId()) == MAX_AIS) {
+				if ((skirmishAIId = ReserveSkirmishAIId()) == SPECIAL_AI_ID) {
```
Comment (sprunk) [inline rts/Net/GameServer.cpp:1673]: This one is also about the limit.

**Review by sprunk — CHANGES_REQUESTED** [review]
Doesn't compile.

**Comment by sprunk** [comment]
Any recent advances? (also in #1533)

**Comment by sprunk** [comment]
Looks like no progress is happening.

**Pull request closed without merging**